### ISSUE TITLE
Understanding how to present the user with the ng-message

### ISSUE THREAD

**Issue by peturh** [json-schema-form/angular-schema-form-material#14]
I'm trying to present the user with the error messages below the input fields, just as they do with the bootstrap schema form. But I'm unable to understand this. I can see that we have something called `sf-messages` on the elements, but how do I populate them?

**Comment by Anthropic** [comment]
They will populate just like the bootstrap ones do, there have just been changes in the way AM handles them so I need to do some updates, but the builder essentially uses sf-messages to append the block to each element so we only have to edit it in one spot.

**Comment by peturh** [comment]
```
    function sfMessagesBuilder(args) {
      var messagesDiv = args.fieldFrag.querySelector('[sf-messages]');
      if (messagesDiv && sfMessagesNode) {
        var child = sfMessagesNode.cloneNode();
        messagesDiv.appendChild(child);
      }
    };
```

This is the builder function your talking about?

**Comment by Anthropic** [comment]
Yes that's it, but it adds in code that David put in to override the way AM works by default to keep it compatible. There are LOTS of bugs in AM still, so it is hard to implement, however it looks to be starting to stabilise at least api-wise so I will start working on it more very soon.

**Comment by DennisHalmstad** [comment]
made some test and changes in the ng-message branch it works (needs work) but needs angular dependencies to be 1.4.8
